packages/tempus/src/Tempus_IntegratorObserverBasic_impl.hpp

```diff
@@ -109,6 +109,9 @@ observeEndIntegrator(const Integrator<Scalar>& integrator){
        << "  Total runtime = " << runtime << " sec = "
        << runtime/60.0 << " min\n"
        << std::asctime(std::localtime(&end))
+       << "\niSteps = " << integrator.getSolutionHistory()->getCurrentState()->getIndex()
+       << "\nnFail = " << integrator.getSolutionHistory()->getCurrentState()->getMetaData()->getNRunningFailures()
+       << "\n"
```

Comment:
Can we be more descriptive?
Number of steps taken =
Number of failures        =

Also getIndex returns the state's index, and not the number of steps taken.  For restart, this will not be correct.

Reply:
> Also getIndex returns the state's index, and not the number of steps taken. For restart, this will not be correct.

Do we currently keep the track of the number of steps taken?

Reply:
Actually I do not think we do.  I would suggest a loop counter on the while loop in IntegratorBasic::advanceTime.  This would also be available for Subcycling.

Reply:
I guess I should change it to `Number of Accepted Steps Taken` since that is what I'm really using the index for here. 

But with restart, we will have to make sure the starting index counter is correct.